packages/server/src/comment/comment.model.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, Field, ID, ObjectType } from '@nestjs/graphql';
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import mongoose, { Document } from 'mongoose';
+import { TokenPayload } from 'src/auth/user.dto';
+
+@Schema()
+@ObjectType()
+@Directive('@key(fields: "_id")')
+export class Comment {
+  @Field(() => ID)
+  _id: mongoose.Types.ObjectId;
+
+  @Prop({ required: true })
+  @Field()
+  fileId: string;
+
+  @Prop()
+  @Field()
+  parentId: string;
+
+  @Prop()
+  @Field()
+  user: TokenPayload;
```

Comment:
For working with GraphQL federation, we can essentially reference information that is resolved by another GraphQL server. For example, from Cargo, the permissions model looks like [this](https://github.com/hicsail/object-storage-service/blob/main/packages/server/src/perms/perms.model.ts#L23) where the type is a string (the ID of the user), but the field is annotated as a `UserModel`. Then in the resolver, there is a special function [here](https://github.com/hicsail/object-storage-service/blob/main/packages/server/src/perms/perms.resolver.ts#L84) that tells Apollo Federation how to fully resolve the user model.

There is a bit going on that looks like magic at first glance with Apollo Federation, so let me know if you want to chat about what is happening